agent/src/test/java/com/teamscale/jacoco/agent/AgentHttpServerTest.java

```diff
@@ -0,0 +1,104 @@
+package com.teamscale.jacoco.agent;
+
+import com.teamscale.client.TeamscaleServer;
+import com.teamscale.jacoco.agent.options.AgentOptionParseException;
+import com.teamscale.jacoco.agent.options.AgentOptions;
+import com.teamscale.jacoco.agent.options.TestAgentOptionsBuilder;
+import com.teamscale.jacoco.agent.upload.UploaderException;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AgentHttpServerTest {
+
+	private Agent agent;
+	private final Integer httpServerPort = 8081;
+	private final String defaultCommitMessage = "Some Message";
+	private final String defaultPartition = "Some Partition";
+
+	/** Starts the http server to control the agent */
+	@BeforeEach
+	public void setup() throws AgentOptionParseException, UploaderException {
+		AgentOptions options = new TestAgentOptionsBuilder()
```

Comment:
I generally prefer a `static` `someAgentOptions` factory method, that makes sure that all fields have valid values (so I can immediately call `create` if I don't want to override anything). In this case in particular, I get the feeling that simply the default options are what we want in the test; any customization only suggests that the custom values are somehow important for the test.

Reply:
They are, assuming that we won't only be using the builder for the `AgentHttpServerTest` in the future.
* `withHttpServerPort` is required to be set in order to start the http server
* `withTeamscaleMessage` is relevant for the two message tests
* `withTeamscalePartition` is relevant for the two partition tests
Since we don't need to set any other options for the agent to work, I saw no sense in providing a static factory method.

Reply:
Essentially, the current implementation is a compromise between implementing the builder to support all agent options (which we don't need yet) and implementing it in a way that it only fits the `AgentHttpServerTest` (i.e. setting the required options as default options in a static factory method). This way, we can easily extend it in the future.